src/audio/selector.c

```diff
@@ -0,0 +1,445 @@
+/*
+ * Copyright (c) 2019, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Lech Betlej <lech.betlej@linux.intel.com>
+ */
+
+/**
+ * \file audio/selector.c
+ * \brief Audio channel selection component. In case 1 output channel is
+ * \brief selected in topology the component provides the selected channel on
+ * \brief ouput. In case 2 or 4 channels are selected on output the componet
+ * \brief works in a bypass mode.
+ * \authors Lech Betlej <lech.betlej@linux.intel.com>
+ */
+
+#include <stddef.h>
+#include <errno.h>
+#include <sof/sof.h>
+#include <sof/lock.h>
+#include <sof/list.h>
+#include <sof/stream.h>
+#include <sof/alloc.h>
+#include <sof/work.h>
+#include <sof/clk.h>
+#include <sof/ipc.h>
+#include "selector.h"
+#include <sof/math/numbers.h>
+
+/**
+ * \brief Validates and sets channel count and channel index.
+ * \details if input data is not supported trace error is displayed and 
+ * \details input data is modified to safe values
+ * \param[in,out] cd Selector component private data.
+ * \param[in] in_channels Number of input channels to the module
+ * \param[in] out_channels Number of output channels to the module
+ * \param[in] ch_idx Index of a channel to be provided on output
+ */
+static void sel_set_channel_values(struct comp_data *cd, 
+						uint32_t in_channels, uint32_t out_channels, uint32_t ch_idx)
+{
+	if ((in_channels != SEL_IN_2CH) || (in_channels != SEL_IN_4CH))
+	{
```

Comment:
C-style? brackers are supposed to be at the end of line

Reply:
The inner parentheses are not needed. Didn't the pre- and post-commit check scripts note that?

Reply:
@singalsu I think checkpatch doesn't check that, anyway even when parentheses are not mandatory they improve readability imho. You are sure what order is instead of wondering what operator had what priority ;p

Reply:
breakers corrected,  I left parenthesis in place